lib/spack/spack/cmd/develop.py

```
@@ -0,0 +1,88 @@
+        help='Do not clone. The package already exists at the source path')
+    clone_group.add_argument(
+        '--clone', action='store_true', dest='clone', default=None,
+        help='Clone the package even if the path already exists')
+    arguments.add_common_arguments(subparser, ['spec'])
+
+
+def develop(parser, args):
+    env = ev.get_env(args, 'develop', required=True)
+
+    if not args.spec:
+        if args.clone is False:
+            raise SpackError("No spec provided to spack develop command")
+
+        # download all dev specs
+        for name, entry in env.dev_specs.items():
+            path = entry.get('path', name)
+            abspath = path if os.path.isabs(path) else os.path.join(
+                env.path, path)
+
+            if os.path.exists(abspath):
+                msg = "Skipping developer download of %s" % entry['spec']
+                msg += " because its path already exists."
+                tty.msg(msg)
+                continue
+
+            stage = spack.spec.Spec(entry['spec']).package.stage
+            stage.steal_source(abspath)
+
+        if not env.dev_specs:
+            tty.warn("No develop specs to download")
+
+        return
+
+    specs = spack.cmd.parse_specs(args.spec)
+    if len(specs) > 1:
+        raise SpackError("spack develop requires at most one named spec")
+
+    spec = specs[0]
+    if not spec.versions.concrete:
+        raise SpackError("Packages to develop must have a concrete version")
+
+    # default path is relative path to spec.name
+    path = args.path or spec.name
+
+    # get absolute path to check
+    abspath = path
+    if not os.path.isabs(abspath):
+        abspath = os.path.join(env.path, path)
+
+    # clone default: only if the path doesn't exist
+    clone = args.clone
+    if clone is None:
+        clone = not os.path.exists(abspath)
+
+    if not clone and not os.path.exists(abspath):
+        raise SpackError("Provided path %s does not exist" % abspath)
```

You may also want to throw an error if `args.clone and os.path.exists(abspath)`

---

Yep, and add a `-f` argument to force it (remove what exists in the path before cloning). Good idea.